sdk/agent/events.go

```diff
@@ -0,0 +1,52 @@
+package agent
+
+import "github.com/stellar/experimental-payment-channels/sdk/state"
+
+type Event interface {
+	event()
+}
+
+// ErrorEvent occurs when an error has occurred, and contains the error
+// occurred.
+type ErrorEvent struct {
+	Err error
+}
+
+func (e ErrorEvent) event() {}
+
+// ConnectedEvent occurs when the agent is connected to another participant.
+type ConnectedEvent struct{}
```

Comment:
❓  should we add a `DisconnectedEvent`? I don't _think_ this is covered by the ErrorEvent currently, but if it is maybe we should make it its own event type

Reply:
Probably, but nothing triggers it at the moment. There are a few edge conditions that aren't covered in the agent yet that need plugging. I created an issue for this at #264.